### PR TITLE
🎯 Bookstore | Add Redux Components🎯

### LINKED ISSUE #4: Add React Components
### Project requirements

- Wrap your `<App>` with the `<Provider>` component from `react-redux`
- Pass your configured `store` into the `<Provider>`
- Create an `initialState` variable for your `booksSlice`, which will include the following:
  ```json
  // Initial state:
  [
    {
      "item_id": "item1",
      "title": "The Great Gatsby",
      "author": "John Smith",
      "category": "Fiction"
    },
    {
      "item_id": "item2",
      "title": "Anna Karenina",
      "author": "Leo Tolstoy",
      "category": "Fiction"
    },
    {
      "item_id": "item3",
      "title": "The Selfish Gene",
      "author": "Richard Dawkins",
      "category": "Nonfiction"
    }
  ]
  ```
- Display your `books`, received from the slice, in a reusable component
- Dispatch actions using `useDispatch`
- Add a `<Button>` component, which includes:
  - An event handler that **adds** a book to the `books` array (with attributes _id_, _title_ and _author_)
- Add a `<Button>` component, which includes:
  - An event handler that **removes** a book from the `books` array (by _id_)

### PR DESCRIPTION
**Pull request by Rysth**
# 🏅 Features Implemented
_Project Requirements -> https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions_v2.md_

List of things that were implemented based on the requirements:

## 📝 General requirements

- ✅ No Linters errors.
- ✅ Used GitFlow correctly.
- ✅ Created a professional `README` file.

## 📝 JavaScript requirements

- ✅ Followed the best JavaScript practices.

## 📝 Project requirements

- ✅ Wrapped the **App** component within a **Provider** component from `react-redux`.
- ✅ Passed the **Store** component to the **Provider**.
- ✅ Changed the `initialState` data to the `books` array from the `booksSlice`.
- ✅ Displayed the `books`, received from the `booksSlice`.
- ✅ Dispatched the actions using the `useDispatch` function from `react-redux`.
- ✅ Added a `Button` component to handle the `addBooks ` reducer from `booksSlice`.
- ✅ Added a `Button` component to handle the `removeBook` reducer from `booksSlice`.

### BooksSlice

books

### Add Book

add

### Remove Book

remove

_John Palacios_
_Microverse Student_

### CODE REVIEW

File: src/components/Book/Book.js [hunk 1]
```diff
@@ -1,31 +1,38 @@
 import React from 'react';
 // eslint-disable-next-line import/no-extraneous-dependencies
 import PropType from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { booksActions } from '../../redux/books/booksSlice';
+import Button from '../Button/Button';
+
+/* eslint-disable */
+function Book({ ID, category, title, author }) {
+  /* eslint-enable */
```
Comment (gemmen29) [inline src/components/Book/Book.js]: - [x] Please remove the `eslint-disable` statements as this defies the purpose of using linters in the first place.

File: src/components/Button/Button.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import PropType from 'prop-types';
+
+/* eslint-disable */
+function Button({ classList, buttonType, text, handleData }) {
+  /* eslint-enable */
```
Comment (gemmen29) [inline src/components/Button/Button.js]: - [x] Please remove the `eslint-disable` statements as this defies the purpose of using linters in the first place.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @Rysth, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Button/Button.js [hunk 3]
```diff
@@ -2,7 +2,15 @@ import React from 'react';
 // eslint-disable-next-line import/no-extraneous-dependencies
 import PropType from 'prop-types';
 
-function Button({ classList, buttonType, text, handleData }) {
+function Button(
+  /* prettier-ignore */
+  {
```
Comment (TadesseDev) [inline src/components/Button/Button.js]: - [x] Dear @Rysth, Just as mentioned by the previous reviewer, please Consider removing disabled liters. Remember linter is here to force you to write clean code and help you identify what could possibly be improved on the current codebase. Please consider removing all the disabled liters.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @Rysth 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

199733072-bdacca19-8df9-43a7-be93-3aff2f1c15c8

 ## Highlights
 - [x] Good job making most of the changes requested by the previous reviewer. 🙏


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TadesseDev — APPROVED** [review]
### Hi @Rysth 


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
AmazingCreepyFrogmouth-max-1mb




## Highlights
 - [x] Good job making the changes. ✅

## Optional suggestions

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Rysth** [comment]
@gemmen29 & @TadesseDev Thank you!

**Pull request merged**